getBlock err 400, but index out of range error occurs

the reason is we didn't handle the getblocks result.Error properly.


```
2018/11/01 18:04:44 Tail: [8ead259949a8562dc39f3cf0092c5262b2e44b0a4dd14879d93ad505f07dbd68](160) type[1]
2018/11/01 18:04:44 Block hash [0564bbc3a0cae6480690addf9b1d0918d5cefd0fbcb76fb48ae9011d83d3ed95], prev[c653288ff754261fcbd7a635e637fe77cdd344ead10047ccea9b995ee61637b9] trigger recovery

2018/11/01 18:04:44 start Fetch blocks
2018/11/01 18:04:44 Tail: [8ead259949a8562dc39f3cf0092c5262b2e44b0a4dd14879d93ad505f07dbd68](160) type[1]
2018/11/01 18:04:44 forked block[8ead259949a8562dc39f3cf0092c5262b2e44b0a4dd14879d93ad505f07dbd68](#160) is detected

2018/11/01 18:04:44 fetch [1] blocks start hash [c653288ff754261fcbd7a635e637fe77cdd344ead10047ccea9b995ee61637b9]
2018/11/01 18:04:44 pack baseMsg.Msg = METHOD
2018/11/01 18:04:44 received buf len [46]
2018/11/01 18:04:44 received baseMsg:  RESULT
2018/11/01 18:04:44 fetch block result error [400]
2018/11/01 18:04:44 consume method finished
panic: runtime error: index out of range

goroutine 13 [running]:
github.com/FissionAndFusion/lws/internal/stream/block.(*BlockFetcher).findForkedBlock(0xc0002837c0, 0x21)
	/Users/soooooot/.go/src/github.com/FissionAndFusion/lws/internal/stream/block/fetch.go:122 +0x165
github.com/FissionAndFusion/lws/internal/stream/block.(*BlockFetcher).checkForkedChain(0xc0002837c0, 0x1)
	/Users/soooooot/.go/src/github.com/FissionAndFusion/lws/internal/stream/block/fetch.go:98 +0x16f
github.com/FissionAndFusion/lws/internal/stream/block.(*BlockFetcher).startFetchBlocks(0xc0002837c0, 0xc0002e04e0)
	/Users/soooooot/.go/src/github.com/FissionAndFusion/lws/internal/stream/block/fetch.go:67 +0x6f

```

--- a/internal/stream/block/fetch.go
+++ b/internal/stream/block/fetch.go
@@ -3,6 +3,7 @@ package block
 import (
 	"bytes"
 	"encoding/hex"
+	"fmt"
 	"log"
 
 	"github.com/FissionAndFusion/lws/internal/coreclient"
@@ -181,7 +182,7 @@ func (b *BlockFetcher) fetch(hash []byte, num int32) ([]*lws.Block, error) {
 	}
 
 	if result.Error != "" {
-		log.Printf("fetch block result error [%s]", result.Error)
+		err = fmt.Errorf("fetch block result error [%s]", result.Error)
 		return nil, err
 	}
 

